src/reducers/index.js

```diff
@@ -144,16 +144,31 @@ export function isFetchingPipelineResources(state) {
 
 export function getPipelineRuns(
   state,
-  { namespace = getSelectedNamespace(state) } = {}
+  { filters, namespace = getSelectedNamespace(state) } = {}
 ) {
-  return pipelineRunsSelectors.getPipelineRuns(state.pipelineRuns, namespace);
+  const runs = pipelineRunsSelectors.getPipelineRuns(
+    state.pipelineRuns,
+    namespace
+  );
+  return runs.filter(pipelineRun => {
+    return filters.every(filter => {
+      const [filterKey, filterValue] = filter.split(':');
+      if (filterKey !== undefined && filterValue !== undefined) {
+        const value = pipelineRun.metadata.labels[filterKey];
+        if (value === filterValue) {
+          return true;
+        }
+      }
+      return false;
```

Comment:
This can be done in one line.

`return filterKey && filterValue && pipelineRun.metadata.labels[filterKey] == filterValue`